CMakeLists.txt

```diff
@@ -56,297 +63,80 @@ zephyr_include_directories(
-  $<$<COMPILE_LANGUAGE:CXX>:-fdata-sections>
-
-  $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
-  $<$<COMPILE_LANGUAGE:ASM>:-D_ASMLANGUAGE>
-)
-
-if(NOT CONFIG_RTTI)
-zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
-)
-endif()
-
-if(NOT CONFIG_EXCEPTIONS)
-zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-fno-exceptions>
-)
-endif()
-
-if(NOT CONFIG_NATIVE_APPLICATION)
-zephyr_ld_options(
-  -nostdlib
-  -static
-  -no-pie
-)
-endif()
-
-if(CONFIG_LIB_CPLUSPLUS)
-zephyr_ld_options(
-  -lstdc++
-)
-endif()
-
-# ==========================================================================
-#
-# cmake -DW=... settings
-#
-# W=1 - warnings that may be relevant and does not occur too often
-# W=2 - warnings that occur quite often but may still be relevant
-# W=3 - the more obscure warnings, can most likely be ignored
-# ==========================================================================
-if(W MATCHES "1")
-  zephyr_compile_options(
-    -Wextra
-    -Wunused
-    -Wno-unused-parameter
-    -Wmissing-declarations
-    -Wmissing-format-attribute
-    -Wold-style-definition
-    )
-  zephyr_cc_option(
-    -Wmissing-prototypes
-    -Wmissing-include-dirs
-    -Wunused-but-set-variable
-    -Wno-missing-field-initializers
-    )
-endif()
-
-if(W MATCHES "2")
-  zephyr_compile_options(
-    -Waggregate-return
-    -Wcast-align
-    -Wdisabled-optimization
-    -Wnested-externs
-    -Wshadow
-    )
-  zephyr_cc_option(
-    -Wlogical-op
-    -Wmissing-field-initializers
-    )
-endif()
-
-if(W MATCHES "3")
-  zephyr_compile_options(
-    -Wbad-function-cast
-    -Wcast-qual
-    -Wconversion
-    -Wpacked
-    -Wpadded
-    -Wpointer-arith
-    -Wredundant-decls
-    -Wswitch-default
-    )
-  zephyr_cc_option(
-    -Wpacked-bitfield-compat
-    -Wvla
-    )
-endif()
+# Discover if the default C standard has been overridden
+# So far, only POSIX-based boards does this
+get_property(CSTD GLOBAL PROPERTY CSTD)
+set_ifndef(CSTD c99)  # Set the default C standard to ISO/IEC 9899:1999
+
+
+## @Brief: toolchain_cc: Macro that expands into compiler options.
+## @Brief: toolchain_ld: Macro that expands into linker options.
+## @Details:
+##   toolchain_cc must have been defined by compiler/${COMPILER}.cmake, where
+##     COMPILER was set by toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}.cmake.
+##   toolchain_ld must have been defined by linker/${LINKER}.cmake, where
+##     LINKER was set by toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}.cmake.
+##
+##   We can classify compiler/linker options into 4 groups:
+##
+##     1. Toolchain-specific and Zephyr-agnostic:
+##          Covers: ABI, call-convention, code generation options, etc.
+##          Where: Defines and includes should be should be captured by
+##            {TOOLCHAIN_C_FLAGS, TOOLCHAIN_INCLUDES} set in ${COMPILER}.cmake.
+##            Compiler options that are not defines or includes, should be set
+##            directly in ${COMPILER}.cmake or ${LINKER}.cmake.
```

Comment:
I'm not able to follow what the "Where" section is trying to say here.

Reply:
It means the physical location of these options.